androidHyperskillApp/src/main/java/org/hyperskill/app/android/step_quiz/view/fragment/DefaultStepQuizFragment.kt

```diff
@@ -322,23 +345,27 @@ abstract class DefaultStepQuizFragment :
     }
 
     override fun render(state: StepQuizFeature.State) {
-        viewStateDelegate?.switchState(state)
+        stepQuizStateDelegate?.switchState(state.stepQuizState)
 
         val stepQuizButtonsLinearLayout = view?.findViewById<LinearLayout>(R.id.stepQuizButtons)
         if (stepQuizButtonsLinearLayout != null) {
             for (childView in stepQuizButtonsLinearLayout.children) {
-                childView.isEnabled = !StepQuizResolver.isQuizLoading(state)
+                childView.isEnabled = !StepQuizResolver.isQuizLoading(state.stepQuizState)
             }
         }
 
-        when (state) {
-            StepQuizFeature.State.Unsupported -> {
+        when (state.stepQuizState) {
+            StepQuizFeature.StepQuizState.Unsupported -> {
                 stepQuizFeedbackBlocksDelegate?.setState(StepQuizFeedbackState.Unsupported)
             }
-            is StepQuizFeature.State.AttemptLoaded -> {
+            is StepQuizFeature.StepQuizState.AttemptLoaded -> {
                 setStepHintsFragment(step)
-                setProblemsLimitFragment(stepRoute)
-                renderAttemptLoaded(state)
+                (state.stepQuizState as? StepQuizFeature.StepQuizState.AttemptLoaded)?.let { attemptLoadedState ->
```

Comment:
It's better to declare `stepQuizState` in `when` expression to avoid cast.
`when (val stepQuizState = state.stepQuizState)`.